x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/PredictionFieldType.java

```diff
@@ -57,6 +58,19 @@ public Object transformPredictedValue(Double value, String stringRep) {
                 }
                 return areClose(value, 1.0D);
             case NUMBER:
+                if (Strings.isNullOrEmpty(stringRep)) {
+                    return value;
+                }
+                // Quick check to verify that the string rep is LIKELY a number
+                // Still handles the case where it throws and then returns the underlying value
+                if (stringRep.charAt(0) == '-' || Character.isDigit(stringRep.charAt(0))) {
```

Comment:
Could we just drop it and rely solely on `parseLong`?
Or is this some kind of perf optimization? If so, are you sure it is really faster?

Reply:
It is way faster if `stringRep` is consistently not a number (which could easily be the case). Every call that it is not a number throws an exception, and the JVM has to build a trace.